src/cascadia/QueryExtension/OpenAILLMProvider.h

```diff
@@ -38,6 +38,7 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
         winrt::hstring _AIKey;
         winrt::Windows::Web::Http::HttpClient _httpClient{ nullptr };
         IBrandingData _brandingData{ winrt::make<OpenAIBranding>() };
+        winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Windows::Web::Http::HttpResponseMessage, winrt::Windows::Web::Http::HttpProgress> _lastRequest{ nullptr };
```

Comment:
You can just make this a local variable that you capture in the `cancellationToken` lambda. This avoids cancelling the wrong request.